### PR TITLE
OCPBUGS-2991: Bump openshift/kubernetes to latest master v2

### PR DESCRIPTION
**Pull request by vrutkovs**
https://github.com/openshift/origin/pull/27548 + bug fix

### CODE REVIEW

File: test/extended/templates/template.go [hunk 1]
```diff
@@ -51,7 +52,7 @@ var _ = g.Describe("[sig-devex][Feature:Templates] template-api", func() {
 				Raw: []byte(runtime.EncodeOrDie(corev1Codec, &v1.Service{
 					ObjectMeta: metav1.ObjectMeta{
 						Name:      "${NAME}-tester",
-						Namespace: "somevalue",
+						Namespace: oc.Namespace(),
```
Comment (deads2k) [inline test/extended/templates/template.go:55]: is this the namespace you needed to change or the on new line 45?

**Comment by vrutkovs** [comment]
Failing tests:

[sig-arch] Managed cluster should ensure platform components have system-* priority class associated [Suite:openshift/conformance/parallel]

/test e2e-gcp-ovn

**Comment by vrutkovs** [comment]
/test e2e-aws-ovn-fips

infra issue

File: test/extended/templates/template.go (hunk 1 above)
Comment (deads2k) [inline test/extended/templates/template.go:55]: This line change is still a problem.  We don't want to mutate this test fixture.

File: test/extended/templates/template.go [hunk 2]
```diff
@@ -124,7 +125,11 @@ var _ = g.Describe("[sig-devex][Feature:Templates] template-api", func() {
 				t.Errorf("%q: %v", path, err)
 				return
 			}
-			processedList, err := templateprocessingclient.NewDynamicTemplateProcessor(dynamicClient).ProcessToListFromUnstructured(template.(*unstructured.Unstructured))
+			unstructuredTemplate := template.(*unstructured.Unstructured)
+			if len(unstructuredTemplate.GetNamespace()) == 0 {
```
Comment (deads2k) [inline test/extended/templates/template.go:129]: this entire commit shouldn't be needed after https://github.com/openshift/library-go/pull/1434

**Pull request merged**